### PR TITLE
Patch: Adds public to PatchOperation<T> class for testing

### PR DESCRIPTION
**Pull request by decodingahmed**
## Description

The `PatchOperation<T>` object is now made public.  This will allow consumers to cast any `PatchOperation` objects to the `PatchOperation<T>` class in order to access the `Value` property. 

For the author's use-case this will allow tests to check that the service level code is passing correct values into the patch request.

In order to avoid introducing a breaking change, these changes do not change the return type of the `Add`, `Set`, `Replace` and `Increment` methods of the `PatchOperation` class.

This PR addresses: https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3374

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Closing issues
Closes https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3374

### CODE REVIEW

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs [hunk 1]
```diff
@@ -2,10 +2,11 @@
 // Copyright (c) Microsoft Corporation.  All rights reserved.
 //------------------------------------------------------------
 
-namespace Microsoft.Azure.Cosmos.Tests
+namespace Microsoft.Azure.Cosmos.Tests.Patch
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs:5]: Can we keep the same namespace as before?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperation{T}Tests.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+﻿//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos.Tests.Patch
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperation{T}Tests.cs:5]: Can we keep the same namespace as the other test files? `Microsoft.Azure.Cosmos.Tests`

**Review by ealsur — CHANGES_REQUESTED** [review]

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperation{T}Tests.cs (hunk 2 above)
Comment (decodingahmed) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperation{T}Tests.cs:5]: Sure no problem.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs (hunk 1 above)
Comment (decodingahmed) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs:5]: Sure no problem

**Review by ealsur — APPROVED** [review]

**Pull request merged**